### PR TITLE
Correct example MessageHandler to use addScheduledTask(...)

### PR DESCRIPTION
**Pull request by joazlazer**
- Uses an anonymous implementation of `Runnable()` and adds it via `addScheduledTask()` to prevent threading errors
- Updates the warning block to indicate the changes
- Changes `Items.diamond` to its correct name, `Items.DIAMOND`

**Justification**: What is the point of doing something incorrectly just to correct it a few lines down? Docs should show only the correct implementations as examples.

**_Side question_**: Should the documentation use an anonymous `Runnable` implementing class (as is used with this PR) or with a _block lambda expression_? (**lines 65-70** would instead read as follows)

```java
serverPlayer.getServerWorld().addScheduledTask(() -> {
    serverPlayer.inventory.addItemStackToInventory(new ItemStack(Items.DIAMOND, amount));
});
```

### CODE REVIEW

**Comment by PaleoCrafter** [comment]
I'd say go with a lambda. We're "officially" allowed to use Java 8 features in mods as of 1.12, so we might as well take advantage of that.

File: docs/networking/simpleimpl.md [hunk 1]
```diff
@@ -61,7 +61,13 @@ public class MyMessageHandler implements IMessageHandler<MyMessage, IMessage> {
     EntityPlayerMP serverPlayer = ctx.getServerHandler().playerEntity;
     // The value that was sent
     int amount = message.toSend;
-    serverPlayer.inventory.addItemStackToInventory(new ItemStack(Items.diamond, amount));
+    // Excecute the action on the main server thread by adding it as a scheduled task
```
Comment (sjx233) [inline docs/networking/simpleimpl.md]: typo

**Comment by joazlazer** [comment]
I updated it to use the lambda expression as suggested by Paleo.

**Comment by tterrag1098** [comment]
Thank you for the update, the docs for this page are quite old and were written when 1.8+ was much newer. This makes much more sense now.

**Pull request merged**